graphix/transpiler.py

```diff
@@ -33,6 +36,104 @@ def __init__(self, width):
+        """method to summarize all parameterized gate instructions
+        informs user about all Parameter variables in the circuit
+
+        Parameters
+        ----------
+        store_data : bool
+                    if True saves all information in respective class attribrutes
+        return_data : bool
+                    if True then returns
+                    set - of all the parameters used in the circuit
+
+        """
+
+        if self.num_parameterized_gates == 0:
+            raise ValueError(" no paramaters to assign ")
+
+        elif self.num_parameterized_gates > 0:
+
+            parameterized_gate_indx = self._find_parameterized_gates()
+            parameters = set()
+            for gate_indx in parameterized_gate_indx:
+
+                gate_instr = self.instruction[gate_indx]
+                parameters = parameters.union(gate_instr[-1].parameters)
+
+            if len(parameters) == 0:
+                self.num_parameterized_gates = 0
+
+            if store_data:
+                self.parameters = parameters
+
+            if return_data:
+                return parameters
+
+    def assign_parameters(self, parameter_assignment: dict, verbose: bool = True, inplace: bool = True):
+        """assign numerical values to the parameters in the measurement commands
+
+        Parameters
+        ----------
+        parameter_assignment : dict
+                            dictionary with parameters as keys and numerical assignments as values. eg {param_1: 0.3, params_2: 0.5, ..}
+        verbose  : bool
+                    if True prints status of all assigned and unassigned gate parameters
+        inplace : bool
+                    if True, assigns paramters to the current circuit, otherwise returns a new Circuit with gate parameters assigned.
+
+        """
+
+        if self.num_parameterized_gates == 0:
+            raise ValueError(" no paramaters to assign ")
+
+        parameterized_gate_indx = self._find_parameterized_gates()
+
+        if inplace:
+
+            for gate_indx in parameterized_gate_indx:
+
+                gate_instr = self.instruction[gate_indx]
+                if isinstance(gate_instr[-1], (Parameter)):
+                    gate_instr[-1] = gate_instr[-1].bind(parameter_assignment, allow_unknown_parameters=True).value
+                    print(gate_instr)  ##checkflag
+                    self.instruction[gate_indx] = gate_instr
+
+            self.inspect_parameterized_gates()
+
+        else:
+
+            new_pattern = deepcopy(self)
+
+            for gate_indx in parameterized_gate_indx:
+
+                gate_instr = new_pattern.instruction[gate_indx]
+                if isinstance(gate_instr[-1], (Parameter)):
+                    gate_instr[-1] = gate_instr[-1].bind(parameter_assignment, allow_unknown_parameters=True).value
+                    new_pattern.instruction[gate_indx] = gate_instr
+
+            new_pattern.inspect_parameterized_gates()
```

Comment:
Code duplication that could be avoided by just assigning `self.instruction[gate_indx]` or `new_pattern.instruction[gate_indx]` depending on the `inplace` condition.